src/controllers/controller.bar.js

```diff
@@ -154,7 +160,7 @@ module.exports = function(Chart) {
 			var vscale = me.getValueScale();
 			var base = vscale.getBasePixel();
 			var horizontal = vscale.isHorizontal();
-			var ruler = me._ruler || me.getRuler();
+			var ruler = (!me._ruler || index >= me._ruler.length) ? me.getRuler() : me._ruler;
```

Comment:
Do you think it is a good idea to remove the ruler completely and to let `calculateBarIndexPixels()` calculate the bar size on the fly? That's because the bar size needs to be recalculated every time the position of the bar or the next bars is changed and there seems little reason to cache the recalculated information.